backend-api/src/functions/services/session/tests/sessionService.test.ts

```diff
@@ -0,0 +1,304 @@
+import {
+  ConditionalCheckFailedException,
+  DynamoDBClient,
+  DynamoDBClientResolvedConfig,
+  PutItemCommand,
+  QueryCommand,
+  ServiceInputTypes,
+  ServiceOutputTypes,
+} from "@aws-sdk/client-dynamodb";
+import { AwsStub, mockClient } from "aws-sdk-client-mock";
+import { ISessionService, SessionService } from "../sessionService";
+
+describe("Session Service", () => {
+  let sessionService: ISessionService;
+  let dynamoDbMockClient: AwsStub<
+    ServiceInputTypes,
+    ServiceOutputTypes,
+    DynamoDBClientResolvedConfig
+  >;
+
+  beforeEach(() => {
+    sessionService = new SessionService("mockTableName");
+    dynamoDbMockClient = mockClient(DynamoDBClient);
+  });
+
+  beforeAll(() => {
+    jest.useFakeTimers().setSystemTime(new Date("2024-03-10"));
+  });
+
+  describe("Get active session ID", () => {
+    describe("Given an error happens when fetching an active session from the table", () => {
+      it("Returns an error response", async () => {
+        dynamoDbMockClient.on(QueryCommand).rejectsOnce("Mock DB Error");
+
+        const result = await sessionService.getActiveSessionId("mockSub");
+
+        expect(result.value).toStrictEqual({
+          errorMessage: "Error getting session ID - Error: Mock DB Error",
+          errorCategory: "SERVER_ERROR",
+        });
+        expect(result.isError).toBe(true);
+      });
+    });
+
+    describe("Given the query response does not contain the Items field", () => {
```

Comment:
Is this a server error? Trying to work out what this means technically - sorry!

Reply:
I think we are looking at these tests again simply because the code has been touched - they are pretty much the same in main and I may just have renamed them.

I don't think it's a server error. DynamoDB is not throwing an exception. 

`Items` is an array that may be empty if nothing was found; however, there is this: ![Screenshot 2024-10-11 at 15 37 10](https://github.com/user-attachments/assets/f456088d-aff7-4fc9-adbf-c3d6351e3ab6)

The type indicates that `Items` could be `undefined` so we need to check that it is not undefined in order to proceed. I don't know why this potential undefined exists and I don't know what it means. I have looked into this briefly in the past and didn't find anything obvious. I think the official documentation is very explicit about Items being an array so I don't think it will ever be undefined, but I don't know...

Reply:
I find sometimes the AWS optional parameters in the SDK responses very unhelpful and unclear.

Reply:
I think we are looking at these tests again simply because the code has been touched - they are pretty much the same in main and I may just have renamed them.

Yep agreed - it's a large refactor so everything sort of comes under scrutiny!

Good to resolve, and thank you for digging into the schema.